tools/storyteller/StoryWriter.java

```diff
@@ -18,70 +16,60 @@
 import javax.inject.Inject;
 import javax.inject.Singleton;
 import tools.storyteller.Protos.Config;
-import tools.storyteller.Protos.FileData;
-import tools.storyteller.Protos.ScreenshotMetadata;
+import tools.storyteller.service.Protos.Story;
+import tools.storyteller.service.Protos.StoryItem;
+import tools.storyteller.service.Protos.StoryList;
 
 
 /* Storyteller writer that writes story files. */
 @Singleton
 public class StoryWriter {
   private static final Logger log = Logger.getForClass();
-  public static DateFormat DATE_FORMATTER = new SimpleDateFormat("yyyy-MM-dd_HH:mm:ss_z");
+  private static final DateFormat DATE_FORMATTER = new SimpleDateFormat("yyyy-MM-dd_HH:mm:ss_z");
 
   private Config config;
   private FileUtils fileUtils;
+  private Story.Builder storyBuilder;
+  private StoryList.Builder storiesBuilder;
+  private long timeMsLastSavedStoryItem;
 
   @Inject
   public StoryWriter(StorytellerConfig storytellerConfig, FileUtils fileUtils) {
     this.config = storytellerConfig.getConfig();
     this.fileUtils = fileUtils;
+    storyBuilder = Story.newBuilder();
+    storiesBuilder = StoryList.newBuilder();
   }
 
-  private String getSharedStoriesPath() {
-    return Storyteller.getSharedStoriesPath(config);
-  }
-
-  private String getUnsharedStoriesPath() {
-    return Storyteller.getUnsharedStoriesPath(config);
-  }
-
-  public void writeStatusFile(FileData.Type type, Message contents) {
+  public void writeStory(Storyteller.StorytellerStatus status, String project) {
     fileUtils.mkdirs(getUnsharedStoriesPath());
-    // We shift by a second to screenshots will be within START and END.
     int secondsShift = 0;
-    if (type == FileData.Type.START) {
-      secondsShift = -1;
-    }
-    if (type == FileData.Type.END) {
-      secondsShift = 1;
-    }
-
-    String timeString = getCurrentTimeString(secondsShift);
-    String filename = timeString + "." + type;
-    try {
-      File statusFile = Paths.get(getUnsharedStoriesPath(), filename).toFile();
-      if (contents == null) {
-        statusFile.createNewFile();
-      } else {
-        fileUtils.writePrototxt(contents, statusFile.toString());
+    switch (status) {
+      case START: {
+        secondsShift = -1;
```

Comment:
I think you should reset with `storyBuilder = Story.newBuilder();`, otherwise you get the leftovers from the previous story.